packages/world/src/systems/CraftSystem.sol

```diff
@@ -20,38 +20,33 @@ import { addToInventoryCount, removeFromInventoryCount, removeAnyFromInventoryCo
 import { requireValidPlayer, requireInPlayerInfluence } from "../utils/PlayerUtils.sol";
 import { notify, CraftNotifData } from "../utils/NotifUtils.sol";
 import { energyToMass, transferEnergyToPool } from "../utils/EnergyUtils.sol";
-import { hashInputs } from "../utils/RecipeUtils.sol";
 import { EntityId } from "../EntityId.sol";
 import { PLAYER_CRAFT_ENERGY_COST } from "../Constants.sol";
 
 contract CraftSystem is System {
-  function craft(EntityId stationEntityId, ObjectTypeId[] memory inputTypes, uint16[] memory inputAmounts) public {
-    require(inputTypes.length > 0, "Recipe not found");
+  function craftWithStation(bytes32 recipeId, EntityId stationEntityId) public {
+    RecipesData memory recipeData = Recipes._get(recipeId);
+    require(recipeData.inputTypes.length > 0, "Recipe not found");
 
     (EntityId playerEntityId, VoxelCoord memory playerCoord, ) = requireValidPlayer(_msgSender());
-
-    requireInPlayerInfluence(playerCoord, stationEntityId);
-
-    EntityId baseStationEntityId = stationEntityId.baseEntityId();
-
-    ObjectTypeId stationObjectTypeId = ObjectType._get(baseStationEntityId);
-
-    bytes32 recipeId = hashInputs(stationObjectTypeId, inputTypes, inputAmounts);
-
-    RecipesData memory recipeData = Recipes._get(recipeId);
+    if (!recipeData.stationTypeId.isNull()) {
+      require(stationEntityId.exists(), "This recipe requires a station");
+      require(ObjectType._get(stationEntityId.baseEntityId()) == recipeData.stationTypeId, "Invalid station");
```

Comment:
is it required to use the baseEntityId? as we are setting it for all entities

Reply:
true! removed it